packages/model/src/data-providers.ts

```diff
@@ -16,26 +16,34 @@ export type FetchCalendarProviderOptions = {
   resourceId?: string;
 };
 
-export interface CalendarDataProvider<
-  CalendarDataObject = DefaultCalendarDataObject,
-  ListCalendarDataObject = DefaultListCalendarDataObject,
+export interface DataProvider<
+  TItemResponse = null,
+  TFetchOptions = null,
+  TListItemResponse = null,
+  TCreateData = null,
```

Comment:
not sure if that works but maybe we should give it a `never` default which means that if its not given you will not be able to type anything into the create method? 
Same for update data maybe

Reply:
i will change it to never and update the partial of create